setup.py

```diff
@@ -5,13 +5,12 @@
 # from flytekit.tools.lazy_loader import LazyLoadPlugin  # noqa
 # extras_require = LazyLoadPlugin.get_extras_require()
 
-MIN_PYTHON_VERSION = (3, 7)
+MIN_PYTHON_VERSION = (3, 6)
 CURRENT_PYTHON = sys.version_info[:2]
 if CURRENT_PYTHON < MIN_PYTHON_VERSION:
     print("Minimum Python version is {}, but you are on {}".format(MIN_PYTHON_VERSION, CURRENT_PYTHON))
```

Comment:
```suggestion
    if CURRENT_PYTHON == (3, 6):
         print(f"Flytekit native typed API is supported for python versions {MIN_PYTHON_VERSION}+, Python 3.6 is supported only for legacy Flytekit API. This will be deprecated soon, we recommend migrating to the new API")
    else:
        print(f"Flytekit API is only supported for Python version is {MIN_PYTHON_VERSION}+. Detected you are on {CURRENT_PYTHON}, installation will not proceed!")
        sys.exit(-1)
```

Reply:
modified slightly and pushed